Providers/nxOMSAutomationWorker/automationworker/3.x/worker/job.py

```diff
@@ -185,6 +188,45 @@ def unload_job(self):
                                     "Unknown",  # TODO(dalbe): fix runbook source
                                     self.runbook.definition_kind_str, self.runbook_data.name, self.job_data.run_on)
         tracer.log_sandbox_job_unloaded(self.job_id)
+    
+    def __process_job_parameters(self):
+        string_parameter_type = 'System.String'
+        string_array_parameter_type = 'System.String[]'
+        bool_parameter_type = 'System.Boolean'
+
+        if self.job_data.parameters is not None and len(self.job_data.parameters) > 0:
+            for parameter in self.job_data.parameters:
+                parameter_type = self.__get_parameter_type(str(parameter["Name"]))
+                if (parameter["Value"] is not None):
+                    parameter_value = str(parameter["Value"])
+                    if string_parameter_type.lower() == parameter_type.lower():
```

Comment:
Consider using an inbuilt method called isInstance(), this is used in multiple locations throughout the project. 
https://docs.python.org/3/library/functions.html#isinstance

Reply:
IsInstance() could not be used to check parameter type if I am correct because the parameter values are all coming as strings.  We are identifying the type by mapping job parameters to runbook parameters (which stores type) using parameter name.

Reply:
Makes sense. Also, we are always going to pass this as a string to the runbook so this shouldn't be an issue.